cmake/genny_subdir.cmake

```diff
@@ -0,0 +1,223 @@
+# Copyright 2019-present MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+##
+# Defines the genny_subdir function
+##
+
+#
+# Usage:
+#
+#     genny_subdir(
+#       NAME   project_name
+#       TYPE   SHARED|STATIC|INTERFACE
+#       DEPENDS
+#         list of other libs to link to for output lib
+#       TEST_DEPENDS
+#         list of other libs to link to for test binary
+#       EXECUTABLE name of file to produce for src/main.cpp
+#     )
+#
+# Notes:
+#
+# -   Assumes public headers are in `include` and end with `.hpp`
+# -   Assumes source files are in `src` and end with `.cpp`.
+# -   The file `src/main.cpp` is special. It is not compiled into the
+#     produced library. It is only compiled if the `EXECUTABLE` argument
+#     is given.
+# -   Assumes regular tests are `test/*.cpp` and get parsed by
+#     `ParseAndAddCatchTests`. The test binary is named
+#     `project_name_test`.
+# -   Assumes benchmark tests are `benchmark/*.cpp` and get parsed by
+#     `ParseAndAddCatchTests`. The benchmark test binary is named
+#     `project_name_benchmark`.
+# -   Installs the library to the `GennyLibraryConfig` export
+#
+function(GENNY_SUBDIR)
```

Comment:
maybe this file and function should be called `GennySubdir` to be consistent with our other `.cmake` files?

Either way I don't really like "subdir" being part of the name. Cmake calls each dir a project but `GennyProject` seemed wrong and `GennySubProject` feels incorrect and contrived...open to ideas about other names though.

Reply:
Inspired by your suggestion - renamed to `CreateGennyTargets`